File: aula_0817/Recursao02.java
import javax.imageio.ImageIO;

public class Recursao02 {

    public static void main(String[] args) {

        int parametro = Integer.parseInt(args[0]);
        int vetor[] = new int [parametro];
        for(int i=0; i<vetor.length; i++)
            vetor[i]=i+1;
        imprime(vetor);

        //System.out.println(vetor);
        inversaoRecursivaB(vetor, 0, parametro-1);
        imprime(vetor);

    }

    private static void imprime(int [] vet){
        for(int i : vet)
            System.out.println(i);
        System.out.println("-=-=-=-=-");
    }

    private static void inversaoRecursiva(int vet[]){
        inversaoRecursiva(vet, 0);
    }

    private static void inversaoRecursiva(int vet[], int indice){

        if(indice<vet.length/2){
            int aux=vet[indice];
            vet[indice]=vet[(vet.length-1)-indice];
            vet[(vet.length-1)-indice]=aux;
            inversaoRecursiva(vet, indice+1);
        }


    }
    
    private static void inversaoRecursiva(int vet[], int indice, int tamVet){

        if(indice<tamVet/2){
            int aux=vet[indice];
            vet[indice]=vet[(tamVet-1)-indice];
            vet[(tamVet-1)-indice]=aux;
            inversaoRecursiva(vet, indice+1);
        }


    }
    
    private static void inversaoRecursivaB(int vet[], int idxL, int idxR){

        if(idxL<idxR){
            int aux=vet[idxL];
            vet[idxL]=vet[idxR];
            vet[idxR]=aux;
            inversaoRecursivaB(vet, idxL+1, idxR-1);
        }


    }
    
}

File: aula_0817/Recursao02.c
#include <stdio.h>
#include <stdlib.h>

int inversaoRecursivaA(int*, int, int);
int inversaoRecursivaB(int*, int, int);
void imprime(int*, int);


int main(int argc, char * argv[]){

    int tamVet = atoi(argv[1]);
    int vetor[tamVet];
    for(int i=0; i<tamVet; i++)
        vetor[i]=i+1;

    imprime(vetor, tamVet);
    inversaoRecursivaB(vetor, 0, tamVet-1);
    imprime(vetor, tamVet);


}
void imprime(int *vet, int tamVet){
    for(int i=0; i<tamVet; i++)
      printf("%d\n", vet[i]);
    printf("-=-=-=-=-==-\n");
}

int inversaoRecursivaA(int vet [], int idxL, int idxR){

        if(idxL<idxR){
            int aux=vet[idxL];
            vet[idxL]=vet[idxR];
            vet[idxR]=aux;
            inversaoRecursivaA(vet, idxL+1, idxR-1);
        }
}

int inversaoRecursivaB(int vet[], int indice, int tamVet){

    if(indice<tamVet/2){
        int aux=vet[indice];
        vet[indice]=vet[(tamVet-1)-indice];
        vet[(tamVet-1)-indice]=aux;
        inversaoRecursivaB(vet, indice+1, tamVet);
    }


}
